pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -765,6 +767,22 @@ func (c *Reconciler) createRun(ctx context.Context, rprt *resources.ResolvedPipe
 		},
 	}
 
+	if rprt.PipelineTask.TaskSpec != nil {
+		j, err := json.Marshal(rprt.PipelineTask.TaskSpec.Spec)
```

Comment:
what happens when `spec` is empty here? do we need a check here `rprt.PipelineTask.TaskSpec.Spec`?

Reply:
there is no error for empty spec, in fact a non empty byte sequence is generated. Do you think we should be more defensive here?

Reply:
I am trying to think if there is any use case where the `spec` could be empty, could this ever be used in any way? 

Since this is not producing any error, its safe to go with the way its implemented now and revisit if needed in future.